src/api/goods.ts

```
@@ -8,9 +8,15 @@ export function getAll(): Promise<Good[]> {
 }
 
 export const get5First = () => {
-  return getAll().then(goods => goods); // sort and get the first 5
+  return getAll().then(goods => {
+    const sortedGoods = [...goods].sort((a, b) => a.name.localeCompare(b.name));
+
+    return sortedGoods.slice(0, 5);
+  });
 };
 
 export const getRedGoods = () => {
-  return getAll().then(goods => goods); // get only red
+  return getAll().then(goods => {
```

Same issue for this call site: `getRedGoods` delegates to `getAll()` but there is no guarantee errors will be propagated or handled because `getAll` does not perform error handling. This violates the checklist item #3: **"API requests must include proper error handling (catch errors and handle them in UI/state)."**

Suggested fix: implement error handling inside `getAll` (preferred) or chain a `.catch` here and make sure the UI receives and shows the error. Also ensure non-OK HTTP responses result in a rejected promise so errors are not silently ignored.